### PR TITLE
LIVE-1341: Remove xrp lib

### PR DESCRIPTION
**Pull request by henri-ly**
## Context (issues, jira)

https://ledgerhq.atlassian.net/browse/LIVE-1341

## Description / Usage

Replace Websocket of XRP with JSON-RPC should be more stable cause the websocket was not handle properly.

We should probably retest XRP and all these use case properly

## Expectations

- [ ] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [ ] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

### CODE REVIEW

**Comment by henrily-ledger** [comment]
Still have some work on CI to do

File: src/families/ripple/bridge/js.ts [hunk 1]
```diff
@@ -297,33 +249,36 @@ const txToOperation =
     }
 
     const op: Operation = {
-      id: `${account.id}-${id}-${type}`,
-      hash: id,
+      id: `${account.id}-${hash}-${type}`,
+      hash: hash,
```
Comment (haammar-ledger) [inline src/families/ripple/bridge/js.ts:253]: Have you tried this on an account with existing operations? Won't there be duplicates with different ids? :thinking:

File: src/families/ripple/bridge/js.ts [hunk 2]
```diff
@@ -647,7 +608,9 @@ const prepareTransaction = async (a: Account, t: Transaction) => {
   if (!networkInfo) {
     try {
       const info = await getServerInfo(a.endpointConfig);
-      const serverFee = parseAPIValue(info.validatedLedger.baseFeeXRP);
+      const serverFee = parseAPIValue(
+        String(info.info.validated_ledger.base_fee_xrp)
```
Comment (haammar-ledger) [inline src/families/ripple/bridge/js.ts:612]: `String(null)` will return `"null"`, maybe safer to use `toString`?

File: src/families/ripple/bridge/js.ts [hunk 3]
```diff
@@ -677,7 +640,9 @@ const getTransactionStatus = async (a, t) => {
     feeTooHigh?: Error;
   } = {};
   const r = await getServerInfo(a.endpointConfig);
-  const reserveBaseXRP = parseAPIValue(r.validatedLedger.reserveBaseXRP);
+  const reserveBaseXRP = parseAPIValue(
+    String(r.info.validated_ledger.reserve_base_xrp)
```
Comment (haammar-ledger) [inline src/families/ripple/bridge/js.ts:644]: (also here)

**Review by ghost — COMMENTED** [review]
This is largely a rework of the XRP impl, good job!
I'm thinking it will probably need some non-reg in addition to testing the linked issue...

Also maybe that rework is the occasion to split that huge `js.ts` bridge file into appropriate `js-*.ts` files. But not really mandatory. :smile:

File: src/families/ripple/bridge/js.ts (hunk 1 above)
Comment (haammar-ledger) [reply src/families/ripple/bridge/js.ts:253]: Nevermind I guess id and hash have the same value?

**Comment by henrily-ledger** [comment]
Well I think the issues was related to websocket, since it doesn't open properly so I don't know if I can reproduce it with test right now :/
I'll try to split with js think right now, but will deliver build to QA right now since there's no functional modification by changing this

File: src/families/ripple/bridge/js.ts (hunk 1 above)
Comment (henrily-ledger) [reply src/families/ripple/bridge/js.ts:270]: yeah, they are the same in the previous implem I just reuse it

**Comment by gre** [comment]
I think we need to set up a bot for it and do our own QA first. 🙏

**Comment by gre** [comment]
@henrily-ledger @haammar-ledger I have changed the base to be `family/ripple` which have a XRP run-per-commit on Mere Denis bot. We can merge to it when you are ready.

**Pull request merged**